managedplugin/plugin.go

```diff
@@ -686,6 +687,51 @@ func (c *Client) Versions(ctx context.Context) ([]int, error) {
 	return res, nil
 }
 
+func (c *Client) FindLatestPluginVersion(ctx context.Context, typ PluginType) (string, error) {
+	if c.config.Registry != RegistryCloudQuery {
+		return "", fmt.Errorf("plugin registry is not cloudquery; cannot find latest plugin version")
+	}
+
+	pathSplit := strings.Split(c.config.Path, "/")
+	if len(pathSplit) != 2 {
+		return "", fmt.Errorf("invalid cloudquery plugin path: %s. format should be team/name", c.config.Path)
+	}
+	org, name := pathSplit[0], pathSplit[1]
+
+	ops := HubDownloadOptions{
+		AuthToken:     c.authToken,
+		TeamName:      c.teamName,
+		LocalPath:     c.LocalPath,
+		PluginTeam:    org,
+		PluginKind:    typ.String(),
+		PluginName:    name,
+		PluginVersion: c.config.Version,
+	}
+	hubClient, err := getHubClient(c.logger, ops)
+	if err != nil {
+		return "", fmt.Errorf("failed to get hub client: %w", err)
+	}
+
+	if ops.TeamName == "" {
+		return "", fmt.Errorf("team name is required to find the latest plugin version")
+	}
+
+	resp, err := hubClient.GetPluginWithResponse(ctx, ops.PluginTeam, cloudquery_api.PluginKind(ops.PluginKind), ops.PluginName)
+	if err != nil {
+		return "", fmt.Errorf("failed to get plugin: %w", err)
+	}
+
+	if resp.JSON200 == nil {
+		return "", fmt.Errorf("failed to get latest plugin version: %w", err)
+	}
+
+	if resp.JSON200.LatestVersion == nil {
+		return "", fmt.Errorf("no latest plugin version found")
```

Comment:
Could this be made a silencable error? Maybe return `"", nil`? You could also export an `ErrNoLatestVersion` so that it's more explicit, but it's a very rare scenario (using unpublished plugins...) I don't think a separate error is needed.

Reply:
Since the purpose of this call is to emit a warning (not an error), wouldn't it be better to leave this as is (there wasn't a plugin version, so it makes sense to error), and in the calling code never error, but only emit the warning when we know the plugin is outdated?

Reply:
Done.